### ISSUE TITLE
Race Detection not mentioned in summary, giving confusing results

### ISSUE THREAD

**Issue by howardjohn** [gotestyourself/gotestsum#64]
When `-race` is used an a race is detected, I get

```
=== FAIL: galley/pkg/config/source/kube/fs TestAddUpdateDelete (0.04s)

```
(no info other than FAIL)

But when a test failures I get a message like

```
=== FAIL: galley/pkg/config/source/kube/fs TestAddUpdateDelete (0.00s)
    source_test.go:141: my test failed
```

It may be useful to mention a data race was detected or something here?

**Comment by dnephin** [comment]
Thanks for the repo! This is something that I would like to fix/improve.

**Comment by dnephin** [comment]
Do you have any way to reproduce the issue or examples of how `go test -v` displays the race failures? I'm not sure where  the race is reported in the test2json output.

**Comment by howardjohn** [comment]
Here is one example I have off hand: https://storage.googleapis.com/istio-prow/logs/integ-race-native-tests_istio_postsubmit/14/build-log.txt (search DATA RACE)

I can make a more minimal reproduction in a bit

**Comment by dnephin** [comment]
I believe those failures are coming from a subtest, is that correct?

The underlying issue may be an a bug in `test2json` output. I put some notes into: #66

**Comment by dnephin** [comment]
I was able to reproduce a data race failure, and I saw the output using `short-verbose` format, so I believe that confirms the issue is #66.

**Comment by dnephin** [comment]
[gotestsum v0.4.2](https://github.com/gotestyourself/gotestsum/releases/tag/v0.4.2) includes a work around for the bug in `test2json`. I'm still hoping my patch to fix the problem is accepted in Go before 1.15, but until then hopefully the workaround addresses the problem.

I'll keep #66 open to track the issue.
